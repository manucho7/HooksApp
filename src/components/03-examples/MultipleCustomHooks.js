import React from 'react'
import { useFetch } from '../../hooks/useFetch';
import { useCounter } from '../../hooks/useCounter';
import '../02-useEffect/effects.css';

export const MultipleCustomHooks = () => {

    const { counter, increment } = useCounter(1);

    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`);
// &&= si existe la data, lo siguiente, extrae posicion 0 de la data
//!!doblenegacion = transformar null en un false
    const { author, quote } = !!data && data[0];

    return (
        <div>
            <h1>Breaking Bad Quotes ^_^</h1>
            <hr />

            {
                loading 
                ?
                    (
                        <div className="alert alert-info text-center">
                            loading...
                        </div> 
                    )
                :
                    (
                        <blockquote className="blockquote text-right">
                            <p className="mb-0">{ quote }</p>

                            <footer className="blockquote-footer">
                                 { author }
                            </footer>

                        </blockquote> 
                    )
            }

            <button 
                className="btn btn-primary"
                onClick={ increment }
            >
                Siguiente frase
            </button>

        </div>
    )
}
