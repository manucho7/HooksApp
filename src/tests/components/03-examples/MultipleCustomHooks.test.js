import React from 'react';
import '@testing-library/dom';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';
jest.mock( '../../../hooks/useFetch' );
jest.mock( '../../../hooks/useCounter' );

describe('Pruebas en <MultipleCustomHooks>', () => {

    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    });
    
    test('debe de mostrarse correctamente en el snapshot', () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar la informacion', () => {
        useFetch.mockReturnValue({
            data: [{
                author: 'Manu',
                quote: 'hola mundo'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks /> );
        expect( wrapper.find('.alert').exists() ).toBe(false);
        expect( wrapper.find('.mb-0').text().trim() ).toBe('hola mundo');
        expect( wrapper.find('footer').text().trim() ).toBe('Manu');
    });
})
