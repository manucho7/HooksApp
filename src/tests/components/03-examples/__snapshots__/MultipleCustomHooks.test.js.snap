// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks> debe de mostrarse correctamente en el snapshot 1`] = `
<div>
  <h1>
    Breaking Bad Quotes ^_^
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente frase
  </button>
</div>
`;
